src/xrpld/app/tx/detail/LoanBrokerCoverDeposit.cpp

```diff
@@ -0,0 +1,121 @@
+#include <xrpld/app/tx/detail/LoanBrokerCoverDeposit.h>
+//
+#include <xrpld/app/misc/LendingHelpers.h>
+
+namespace ripple {
+
+bool
+LoanBrokerCoverDeposit::checkExtraFeatures(PreflightContext const& ctx)
+{
+    return checkLendingProtocolDependencies(ctx);
+}
+
+NotTEC
+LoanBrokerCoverDeposit::preflight(PreflightContext const& ctx)
+{
+    if (ctx.tx[sfLoanBrokerID] == beast::zero)
+        return temINVALID;
+
+    auto const dstAmount = ctx.tx[sfAmount];
+    if (dstAmount <= beast::zero)
+        return temBAD_AMOUNT;
+
+    if (!isLegalNet(dstAmount))
+        return temBAD_AMOUNT;
+
+    return tesSUCCESS;
+}
+
+TER
+LoanBrokerCoverDeposit::preclaim(PreclaimContext const& ctx)
+{
+    auto const& tx = ctx.tx;
+
+    auto const account = tx[sfAccount];
+    auto const brokerID = tx[sfLoanBrokerID];
+    auto const amount = tx[sfAmount];
+
+    auto const sleBroker = ctx.view.read(keylet::loanbroker(brokerID));
+    if (!sleBroker)
+    {
+        JLOG(ctx.j.warn()) << "LoanBroker does not exist.";
+        return tecNO_ENTRY;
+    }
+    if (account != sleBroker->at(sfOwner))
+    {
+        JLOG(ctx.j.warn()) << "Account is not the owner of the LoanBroker.";
+        return tecNO_PERMISSION;
+    }
+    auto const vault = ctx.view.read(keylet::vault(sleBroker->at(sfVaultID)));
+    if (!vault)
+    {
+        JLOG(ctx.j.fatal()) << "Vault is missing for Broker " << brokerID;
+        return tefBAD_LEDGER;
+    }
```

Comment:
nit: In other places we exclude such checks for vault from coverage reports.

```suggestion
    {
        // LCOV_EXCL_START
        JLOG(ctx.j.fatal()) << "Vault is missing for Broker " << brokerID;
        return tefBAD_LEDGER;
        // LCOV_EXCL_STOP  
    }
```

Edit: nit